R/utils-ui.R

```diff
@@ -271,3 +271,89 @@ compute_n_show <- function(n, n_show_nominal = 5, n_fudge = 2) {
     n
   }
 }
+
+# menu(), but based on readline() + cli and mockable ---------------------------
+# https://github.com/r-lib/cli/issues/228
+# https://github.com/rstudio/rsconnect/blob/main/R/utils-cli.R
+
+cli_menu <- function(header,
+                     prompt,
+                     choices,
+                     not_interactive = choices,
+                     exit = integer(),
```

Comment:
I called this `exit` (instead of `quit`) to better match the typical vocabulary here (e.g. in the docs for `utils::menu()`) and to avoid any confusion with actuallly quitting R.